Update composite documentation to be more explicit (#11282)
DataDog/documentation: content/en/monitors/monitor_types/composite.md

For detailed instructions on the **Say what's happening** and **Notify your team** sections, see the [Notifications][2] page.

### API

> When using the API, a composite monitor's query is defined in terms of its constituent monitors using monitor IDs.

Reviewer: ```suggestion
When using the API, a composite monitor's query is defined in terms of its constituent monitors using monitor IDs.
```
I am not sure what you mean here by constituent monitors? Dependent monitors? Related monitors?

Author: Used the same word as before but we definitely intent to say "Child monitor" or sub-monitor here.

> When using the API, a composite monitor's query is defined in terms of its constituent monitors using monitor IDs.

Author: ```suggestion
When using the API, a composite monitor's query is defined in terms of its sub-monitors using monitor IDs.
```

### Computing trigger conditions

> There are 4 different statuses a composite monitor and 3 of them are considered as alert-worthy:

Reviewer: ```suggestion
There are 4 different statuses a composite monitor and 3 of them are considered alert-worthy:
```

> There are 4 different statuses a composite monitor considers and 3 of them are considered as alert-worthy:

Reviewer: ```suggestion
There are 4 different statuses a composite monitor and 3 of them are considered as alert-worthy:
```

| Status    | Alert-worthy         |Severity           |
|-----------|----------------------|-------------------|
| `Alert`   | True                 |4 (Most severe)    |
| `Warn`    | True                 |3                  |
| `No Data` | True                 |2                  |
| `Ok`      | False                |1 (Least severe)   |

The Boolean operators used (`&&`, `||`, `!`) operate on the alert-worthiness of the composite monitor status.

* If `A && B` is alert-worthy, the result is the **least** severe status between A and B.
* If `A || B` is alert-worthy, the result is the **most** severe status between A and B.
* If `A` is `No Data`, `!A` is `No Data`
* If `A` is alert-worthy, `!A` is `OK`
* If `A` is no alert-worthy, `!A` is `Alert`

> Consider a composite monitor that uses three individual monitors (`A`, `B`, `C`) and a trigger condition `A && B && C`. The following table shows the resulting status of the composite monitor given different statuses for its individual monitors (alert-worthiness is indicated with T or F):

Reviewer 2: ```suggestion
Consider a composite monitor that uses two individual monitors `A` and `B`. The following table shows the resulting status of the composite monitor given the trigger condition (`&&` or `||`), and the different statuses for its individual monitors (alert-worthiness is indicated with T or F):
```

> Consider a composite monitor that uses two individual monitors `A` and `B`. The following table shows the resulting status of the composite monitor given the trigger condition (`&&` or `||`), and the different statuses for its individual monitors (alert-worthiness is indicated with T or F):

Reviewer: ```suggestion
Consider a composite monitor that uses two individual monitors: `A` and `B`. The following table shows the resulting status of the composite monitor given the trigger condition (`&&` or `||`), and the different statuses for its individual monitors (alert-worthiness is indicated with T or F):
```

| Warn (T)    | Ok (F)      | `A || B`    |                  | Warn (T)         | {{< X >}}        |
| No Data (T) | Warn (T)    | `A && B`    | True             | No Data (T)      | {{< X >}}        |
| No Data (T) | Warn (T)    | `A || B`    | True             | Warn (T)         | {{< X >}}        |
| No Data (T) | Warn (T)    | `A && B`    | False            | Last known       |                  |
| No Data (T) | Warn (T)    | `A || B`    | False            | Warn (T)         | {{< X >}}        |
| No Data (T) | OK (F)      | `A && B`    | False            | OK (F)           |                  |
| No Data (T) | OK (F)      | `A || B`    | False            | Last known       |                  |
| No Data (T) | OK (F)      | `A && B`    | True             | OK (F)           |                  |
| No Data (T) | OK (F)      | `A || B`    | True             | No Data (T)      | {{< X >}}        |
| No Data (T) | No Data (T) | `A && B`    | True             | No Data (T)      | {{< X >}}        |
| No Data (T) | No Data (T) | `A || B`    | True             | No Data (T)      | {{< X >}}        |

> **Note**: When the composite has `notify_no_data` to false and the result of the evaluation of the sub-monitors should end up on a `No Data` status for the composite, the composite uses the last known state instead. 

Reviewer: ```suggestion
**Note**: When the composite has `notify_no_data` to false, and the result of the evaluation of the sub-monitors should end up on a `No Data` status for the composite, the composite uses the last known state instead. 
```
